src/components/ItemsList.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import firebase from '../lib/firebase';
+import { useCollectionData } from 'react-firebase-hooks/firestore';
+
+const ItemsList = () => {
+  const [itemsList, loading, error] = useCollectionData(
+    firebase.firestore().collection('items'),
+    {
+      snapshotListenOptions: { includeMetadataChanges: true },
+    },
+  );
+  // useCollectionData extracts firebase.firestore.QuerySnapshot.docs while useCollection
+  // returns firebase.firestore.QuerySnapshot itself.
+  return (
+    <div>
+      <h2>New List</h2>
+      {loading === true ? <p> Loading... </p> : null}
+      {error === undefined ? null : <p> An error has occurred... </p>}
+      {itemsList === undefined ? null : (
```

Comment:
These ternary operators work perfectly. I just want to introduce another approach called [short-circuit conditionals](https://codeburst.io/javascript-short-circuit-conditionals-bbc13ac3e9eb). The idea is that the `&&` operator can be used to shorten our evaluation with boolean values. `&&` returns true when both operands are truthy and returns false if they are not.

This code can be refactored as follows: 

`{ loading && <p> Loading... </p> } `
`{ error && <p> An error has occurred... </p> } `
 `{itemsList && (`
 
 This is just a preference. Ternary evaluation works perfectly as well. 👍

Reply:
Thanks for the alternative approach. I went for ternary rather than short-circuit conditionals to help those who are reading the code to quickly make out the different values/states in which the variables will be without necessarily having to look them up in the documentation. 

For example `error === undefined ? <expr1> : <expr2 ` makes it clear that one of the possible values of `error` is `undefined`. It would be easier to infer the other  value it can possibly take without necessarily looking at the doc. In this case it could be the error object. However `error && <p> </p>` doesn't tell the reader any possible value `error` might take. It could be any `falsy` value like `null` or `''`. One can only be sure after looking it up in the doc. 

So, I have used conditionals to increase readability of the code since most of us participants are just starting out with firebase/react-firebase-hooks.

Reply:
Absolutely! This conversation will be very valuable for others. Thank you for such a detailed explanation. 👏